python/ray/experimental/channel/torch_tensor_type.py

```diff
@@ -1,65 +1,99 @@
+        do not want this behavior, you should deregister the custom serializer
+        using ray.util.serialization.deregister_serializer(torch.Tensor).
+
+        Args:
+            shape: The expected shape of the torch.Tensor. "auto" (default)
+                means that the shape will be dynamically inferred. For tensors
+                passed via host memory (default), the shape is a hint for the
+                maximum size of the tensor. If a DAG node's returned serialized
+                tensor exceeds this size, the task will error. For tensors
+                passed via NCCL, the returned tensor must *match* the given
+                shape; if it does not match, the task will error.
+            dtype: The expected dtype of the torch.Tensor. Similar to the
+                shape, this may be statically or dynamically declared.
+            transport: "auto" (default) means that tensors will be passed via
+                host memory, using numpy as the serialization format. Pass
+                TorchTensorType.NCCL or "nccl" to use NCCL instead, avoiding
+                the host memory copy.
+        """
+        super().__init__()
+
+        if isinstance(shape, str):
+            shape = shape.lower()
+        if isinstance(dtype, str):
+            dtype = dtype.lower()
         self.shape = shape
         self.dtype = dtype
         self.transport = transport
+        self.nccl_group_id: Optional[str] = None
+
+    def register_custom_serializer(self) -> None:
+        super().register_custom_serializer()
 
-    @staticmethod
-    def register_custom_serializer(outer: Any) -> None:
-        # Helper method to run on the DAG driver and actors to register custom
-        # serializers.
         import torch
 
-        from ray.air._internal import torch_utils
+        default_device = _get_default_torch_device()
+        ctx = ChannelContext.get_current()
+        ctx.serialization_context.set_torch_device(default_device)
 
-        if ray.get_gpu_ids():
-            default_device = torch_utils.get_devices()[0]
-        else:
-            # torch_utils defaults to returning GPU 0 if no
-            # GPU IDs were assigned by Ray. We instead want
-            # the default to be CPU.
-            default_device = torch.device("cpu")
+        def serialize(t):
+            ctx = ChannelContext.get_current()
+            return ctx.serialization_context.serialize_tensor(t)
 
-        torch_tensor_serializer = _TorchTensorSerializer(default_device)
+        def deserialize(b):
+            ctx = ChannelContext.get_current()
+            return ctx.serialization_context.deserialize_tensor(b)
 
-        CUSTOM_SERIALIZERS = (
-            (
-                _TorchTensorWrapper,
-                torch_tensor_serializer.serialize_to_numpy,
-                torch_tensor_serializer.deserialize_from_numpy,
-            ),
+        ray.util.serialization.register_serializer(
```

Comment:
Out of curiosity, what's the scope of the serializer? Is it per Ray job, per core worker process, or per channel? To elaborate, if there are multiple nodes in the DAG that use the same ChannelOutputType (e.g., TorchTensorType), do different worker processes for actor tasks reuse the serializer?